src/Storages/System/StorageSystemTimeZones.cpp

```diff
@@ -0,0 +1,27 @@
+#include "StorageSystemTimeZones.h"
+
+#include <algorithm>
+#include <DataTypes/DataTypeString.h>
+
+
+extern const char * auto_time_zones[];
+
+namespace DB
+{
+NamesAndTypesList StorageSystemTimeZones::getNamesAndTypes()
+{
+    return {
+        {"time_zone", std::make_shared<DataTypeString>()},
+    };
+}
+
+void StorageSystemTimeZones::fillData(MutableColumns & res_columns, const Context &, const SelectQueryInfo &) const
+{
+    std::vector<const char *> time_zones;
```

Comment:
Maybe we can do it in single loop, without temporary array?

Reply:
yes definitely, should be fixed now.